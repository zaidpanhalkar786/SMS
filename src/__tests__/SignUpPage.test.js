import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import SignUpPage from '../Signuppage'

// Mock out all top level functions, such as get, put, delete and post:
// jest.mock('axios')
afterEach(cleanup)

describe('SignUpPage', () => {
  test('renders sign up form', () => {
    const root = renderer.create(<SignUpPage />).toJSON()
    expect(root).toMatchSnapshot()
  })

  it('handles form input changes', () => {
    const component = renderer.create(<SignUpPage />)
    const instance = component.root

    const firstNameInput = instance.findByProps({ placeholder: 'First Name' })
    firstNameInput.props.onChange({ target: { value: 'John' } })
    expect(
      instance.findByProps({ placeholder: 'First Name' }).props.value
    ).toBe('John')

    const lastNameInput = instance.findByProps({ placeholder: 'Last Name' })
    lastNameInput.props.onChange({ target: { value: 'Doe' } })
    expect(instance.findByProps({ placeholder: 'Last Name' }).props.value).toBe(
      'Doe'
    )

    const emailInput = instance.findByProps({ placeholder: 'Email' })
    emailInput.props.onChange({ target: { value: 'test@example.com' } })
    expect(instance.findByProps({ placeholder: 'Email' }).props.value).toBe(
      'test@example.com'
    )

    const mobileNoInput = instance.findByProps({ placeholder: 'Mobile No.' })
    mobileNoInput.props.onChange({ target: { value: '8828262521' } })
    expect(
      instance.findByProps({ placeholder: 'Mobile No.' }).props.value
    ).toBe('8828262521')

    const DesignationInput = instance.findByProps({
      placeholder: 'Designation Level'
    })
    DesignationInput.props.onChange({ target: { value: 'Level 5' } })
    expect(
      instance.findByProps({ placeholder: 'Designation Level' }).props.value
    ).toBe('Level 5')

    const passwordInput = instance.findByProps({
      placeholder: 'Password'
    })
    passwordInput.props.onChange({ target: { value: '1234' } })
    expect(instance.findByProps({ placeholder: 'Password' }).props.value).toBe(
      '1234'
    )

    const confirmPasswordInput = instance.findByProps({
      placeholder: 'Confirm Password'
    })
    confirmPasswordInput.props.onChange({ target: { value: '1234' } })
    expect(
      instance.findByProps({ placeholder: 'Confirm Password' }).props.value
    ).toBe('1234')
  })
  // test('handles sign up submission successfully', () => {
  //   const { getByPlaceholderText, getByText } = render(<SignUpPage />)

  //   const firstNameInput = getByPlaceholderText('First Name')
  //   const lastNameInput = getByPlaceholderText('Last Name')
  //   const emailInput = getByPlaceholderText('Email')
  //   const mobileNoInput = getByPlaceholderText('Mobile No.')
  //   const employeeLevelInput = getByPlaceholderText('Designation Level')
  //   const passwordInput = getByPlaceholderText('Password')
  //   const confirmPasswordInput = getByPlaceholderText('Confirm Password')
  //   const saveButton = getByText('Save')

  // fireEvent.change(firstNameInput, { target: { value: 'John' } })
  // fireEvent.change(lastNameInput, { target: { value: 'Doe' } })
  // fireEvent.change(emailInput, { target: { value: 'test@example.com' } })
  // fireEvent.change(mobileNoInput, { target: { value: '1234567890' } })
  // fireEvent.change(employeeLevelInput, { target: { value: 'Manager's } })
  // fireEvent.change(passwordInput, { target: { value: 'password123' } })
  // fireEvent.change(confirmPasswordInput, { target: { value: 'password123' } })

  // fireEvent.click(saveButton)
})
