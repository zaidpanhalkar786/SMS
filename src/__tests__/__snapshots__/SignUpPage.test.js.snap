// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage renders sign up form 1`] = `
<div
  className="ui center aligned middle aligned grid"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="column"
    style={
      Object {
        "maxWidth": 650,
      }
    }
  >
    <h2
      className="ui center aligned header"
    >
      Sign Up
    </h2>
    <form
      className="ui large form"
      onSubmit={[Function]}
    >
      <div
        className="ui stacked segment"
      >
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
            style={
              Object {
                "fontSize": "23px",
              }
            }
          >
            <input
              aria-describedby={null}
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
            style={
              Object {
                "fontSize": "23px",
              }
            }
          >
            <input
              aria-describedby={null}
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
            style={
              Object {
                "fontSize": "23px",
              }
            }
          >
            <input
              aria-describedby={null}
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
            />
            <i
              aria-hidden="true"
              className="mail icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
            style={
              Object {
                "fontSize": "23px",
              }
            }
          >
            <input
              aria-describedby={null}
              onChange={[Function]}
              placeholder="Mobile No."
              type="tel"
              value=""
            />
            <i
              aria-hidden="true"
              className="phone icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
            style={
              Object {
                "fontSize": "23px",
              }
            }
          >
            <input
              aria-describedby={null}
              onChange={[Function]}
              placeholder="Designation Level"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="briefcase icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid action left icon input"
            style={
              Object {
                "fontSize": "23px",
              }
            }
          >
            <input
              aria-describedby={null}
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
              onClick={[Function]}
            />
            <button
              className="ui icon button"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="eye icon"
                onClick={[Function]}
              />
            </button>
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid action left icon input"
            style={
              Object {
                "fontSize": "23px",
              }
            }
          >
            <input
              aria-describedby={null}
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
              onClick={[Function]}
            />
            <button
              className="ui icon button"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="eye icon"
                onClick={[Function]}
              />
            </button>
          </div>
        </div>
        <button
          className="ui purple massive fluid button"
          onClick={[Function]}
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
